src/main/java/net/minecraftforge/client/event/InputEvent.java

```diff
@@ -225,11 +225,11 @@ public ClickInputEvent(int button, KeyBinding keyBinding, Hand hand)
 
         /**
          * Set to false to disable the hand swing animation.
-         * @throws UnsupportedOperationException If this is not an attack input.
+         * @throws UnsupportedOperationException If this is a pick block input.
          */
         public void setSwingHand(boolean value)
         {
-            if(!isAttack()) throw new UnsupportedOperationException("The hand swing can't be disabled if this is not an attack input.");
+            if(isPickBlock()) throw new UnsupportedOperationException("The hand swing can't be disabled if this is a pick block input.");
```

Comment:
I would argue there is no point punishing modders for this. Just put in the method doc that changing this value has no affect for pick blocks.